service/src/main/java/zone/cogni/asquare/service/index/IndexConfigProvider.java

```diff
@@ -13,45 +15,54 @@
 import zone.cogni.libs.jena.utils.JenaUtils;
 import zone.cogni.libs.sparqlservice.SparqlService;
 
+import javax.annotation.Nullable;
 import java.io.File;
+import java.util.function.BiConsumer;
 import java.util.function.Function;
 
+@Builder
+@AllArgsConstructor
 public class IndexConfigProvider {
 
   private final RdfStoreService rdfStoreService;
   private final SparqlService sparqlService;
   private final Function<ResourceIndex, ApplicationProfile> applicationProfileSupplier;
   private final ElasticStore elasticStore;
+  @Nullable
   private final Function<ResourceIndex, Function<TypedResource, ObjectNode>> facetConversionSupplier;
-
-  public IndexConfigProvider(RdfStoreService rdfStoreService,
-                             Function<ResourceIndex, ApplicationProfile> applicationProfileSupplier,
-                             Elasticsearch7Store elasticStore,
-                             Function<ResourceIndex, Function<TypedResource, ObjectNode>> facetConversionSupplier) {
+  @Nullable
+  private final BiConsumer<ObjectNode, String> postIndexInterceptor;
+
+  public IndexConfigProvider( RdfStoreService rdfStoreService,
+                              Function<ResourceIndex, ApplicationProfile> applicationProfileSupplier,
+                              Elasticsearch7Store elasticStore,
+                             @Nullable Function<ResourceIndex, Function<TypedResource, ObjectNode>> facetConversionSupplier,
+                             @Nullable BiConsumer<ObjectNode, String> postIndexInterceptor) {
```

Comment:
isn't this the `@AllArgsConstructor` ?
and the other constructor is gone, so we still have the backwards compatible issue....

note, as we have a builder now, you can set the other constructors to deprecated and make the `@AllArgsConstructor` private